src/firmware/app/control/wheel_controller.c

```diff
@@ -0,0 +1,99 @@
+#include "firmware/app/control/wheel_controller.h"
+
+#include "firmware/shared/circular_buffer.h"
+
+struct WheelController
+{
+    CircularBuffer_t* previous_command_buffer;
+    CircularBuffer_t* previous_output_sample_buffer;
+    float* command_coefficients;
+    float* output_sample_coefficients;
+    size_t num_command_coefficients;
+    size_t num_output_sample_coefficients;
+};
+
+WheelController_t* app_wheel_controller_create(
+    float* command_coefficients, unsigned int num_command_coefficients,
+    float* output_sample_coefficients, unsigned int num_output_sample_coefficients)
+{
+    // Create the circular buffers that will contain the sample and command history
+    CircularBuffer_t* command_buffer = circular_buffer_create(num_command_coefficients);
+    CircularBuffer_t* samples_buffer =
+        circular_buffer_create(num_output_sample_coefficients);
+
+    // Allocate memory to contain the coefficients of the difference equation
+    float* internal_command_coefficients =
+        (float*)malloc(sizeof(float) * num_command_coefficients);
+    float* internal_samples_coefficients =
+        (float*)malloc(sizeof(float) * num_output_sample_coefficients);
+
+    // Allocate memory on the heap
+    WheelController_t* wheel_controller =
+        (WheelController_t*)malloc(sizeof(WheelController_t));
+
+    // Copy the input coefficients to the internal arrays and
+    // fill the circular buffers with zeros as the initial state
+    for (unsigned int i = 0; i < num_command_coefficients; i++)
+    {
+        internal_command_coefficients[i] = command_coefficients[i];
+        circular_buffer_push(command_buffer, 0.0f);
+    }
+    for (size_t i = 0; i < num_output_sample_coefficients; i++)
```

Comment:
Any particular reason for switching between `unsigned int` and `size_t`?

Reply:
Nope, forgot to change that one. 👍